### ISSUE TITLE
Accessibility Issue in RichText: Screen reader doesn't announce state of text formatting controls whether its selected or not when accessing it through tab.

### ISSUE THREAD

**Issue by shashankpatel1542** [pnp/sp-dev-fx-controls-react#1945]
User Experience:
Screen reader dependent users will be impacted as the Selected state is not defined for Style buttons. Due to this, users will not be able to identify the state of the control. This will confuse them.

Actual Result:
Selected state is not defined for Style buttons. i.e. Bold, Italic, Underline.

Note:
Same issue repro if we unselect button, screen reader doesn't announce any information regarding it.

Expected Result:
Selected state should be defined for Style buttons.
Screen reader should announce "selected" information when a button is selected.

Please see attached screenshot where i have selected Underline button but it doesn't announce whether it has been selected or not and also as you can see bold was already selected earlier and when i navigate to it by pressing tab it doesn't announce the selected state of that button which will confuse screen reader users.

![Image](https://github.com/user-attachments/assets/afe39378-8a93-4dbb-aa4d-e8148383d6aa)

**Comment by michaelmaillot** [comment]
Hi @shashankpatel1542,

Thanks for raising this. We definetely need to be better regarding accessibility on the controls.

Would you like to help us on one of theses issues you've declared so far?

**Comment by shashankpatel1542** [comment]
Hi @michaelmaillot ,

Sure, do let me know if you need more info on this and any other help needed.

**Comment by jtlivio** [comment]
The Bold, Italic and Underline buttons pass `checked` to Fluent UI's `IconButton`
without `toggle`. `aria-pressed` is only rendered when `toggle` is set, so the
state never reaches screen readers.

I can take this if nobody is working on it. The Link button also passes
`checked`, but I'd leave it unchanged unless you'd prefer it included.

**Comment by joelfmrodrigues** [comment]
@jtlivio Many thanks! PR merged and a new beta release should be published in a few minutes
